src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<core.mate.academy.model.Machine> getAll(Class type);
```

Comment:
The `getAll` method should be parameterized to accept a `Class<? extends Machine>` instead of a raw `Class` type. This will ensure type safety and align with the intended design of handling subclasses of `Machine`.